graph/protocol.go

```diff
@@ -166,6 +166,42 @@ func AddToMetadata(protocol string, val float64, code, flags, host string, sourc
 	default:
 		log.Tracef("Ignore unhandled metadata protocol [%s]", protocol)
 	}
+
+	if sourceMetadata != nil {
+		if _, ok := sourceMetadata["outbound"]; !ok {
+			sourceMetadata["outbound"] = map[string]map[string]float64{}
+		}
+
+		if _, ok := sourceMetadata["outbound"].(map[string]map[string]float64)[protocol]; !ok {
+			sourceMetadata["outbound"].(map[string]map[string]float64)[protocol] = map[string]float64{
+				code: val,
+			}
+		} else {
+			if _, ok := sourceMetadata["outbound"].(map[string]map[string]float64)[protocol][code]; !ok {
+				sourceMetadata["outbound"].(map[string]map[string]float64)[protocol][code] = val
+			} else {
+				sourceMetadata["outbound"].(map[string]map[string]float64)[protocol][code] += val
+			}
+		}
+	}
+
+	if destMetadata != nil {
+		if _, ok := destMetadata["inbound"]; !ok {
+			destMetadata["inbound"] = map[string]map[string]float64{}
+		}
+
+		if _, ok := destMetadata["inbound"].(map[string]map[string]float64)[protocol]; !ok {
+			destMetadata["inbound"].(map[string]map[string]float64)[protocol] = map[string]float64{
+				code: val,
+			}
+		} else {
+			if _, ok := destMetadata["inbound"].(map[string]map[string]float64)[protocol][code]; !ok {
+				destMetadata["inbound"].(map[string]map[string]float64)[protocol][code] = val
+			} else {
+				destMetadata["inbound"].(map[string]map[string]float64)[protocol][code] += val
+			}
+		}
+	}
```

Comment:
> I'm not too sure about what's going on here. But if you are trying to gather health rate data during graph generation I'd probably vote that down and suggest instead to wait for the graph to be fully generated, and then convert the rate data (if it exists) to the necessary health format.

It's quite an eyesore right now in the WIP but yes it is gathering raw inbound/outbound metrics for each node that looks something like:
```
inbound: {
  "http": {
      "200": 45.634,
      "503": 1.2
    }
}
```

This can probably be moved to the health appender. There was a reason why I put it here but I can't remember why exactly that was now. Maybe the metrics data on the node wasn't quite suitable for health generation but it's worth trying to move it to the appender and then find out.

Reply:
We'll have to look at this because I'd be surprised if the rates we already have on the graph are not suitable for health.  They may just be in a different format.